x/distribution/client/cli/tx.go

```diff
@@ -167,6 +168,12 @@ $ %s tx distribution withdraw-all-rewards --from mykey
 			}
 
 			chunkSize, _ := cmd.Flags().GetInt(FlagMaxMessagesPerTx)
+			if clientCtx.BroadcastMode != flags.BroadcastBlock && chunkSize > 0 {
+				cmd.Printf("Cannot use broadcast mode %[1]s with %[2]s != 0. Forcing %[2]s to 0",
+					clientCtx.BroadcastMode, FlagMaxMessagesPerTx)
+				chunkSize = 0
```

Comment:
is it worth erroring here? To force the the user to add the `--FlagMaxMessagesPerTx=0` flag when not using broadcast mode block.

Reply:
@amaurym To be honest, I think that would result in a bad UX. As  per [this comment](), the `block` broadcast mode should be disincentivized, and the `sync` and `async` should be used instead. If we require the users to set a flag for the modes `sync` and `async`, that might make them simply use the `block` mode instead. For this reason, I think that changing the value and warning them with that log is sufficient. 

Another thing that could be done is simply set the default value of `FlagMaxMessagesPerTx` to be `0`, and error in case `async` or `sync` are used and that value is not zero.

Reply:
Cool. I'm putting automerge on, thanks!

Reply:
@amaurym I've swapped the behavior by setting the default chunks to `0` and returning an error when `async` or `sync` is used with a `chunks` greater than 0